### PR TITLE
Add metrics in prow controllers

### PR DESCRIPTION
**Pull request by kargakis**
@krzyzacy @cjwagner @stevekuznetsov ptal, this is modeled after how we gather metrics in k8s and origin controllers. My only worry was that this would make switching to reflectors harder but I don't think that's true anymore.

### CODE REVIEW

**Comment by kargakis** [comment]
@krzyzacy you should be able to build pod metrics on top of this

**Comment by kargakis** [comment]
Dropping `agent` axis to halve metric cardinality. During setting up alerts, I noticed that `agent` is redundant since you most likely want to monitor services (which includes an agent that may be sharded across multiple services).

File: prow/cmd/plank/main.go [hunk 1]
```diff
@@ -112,6 +115,16 @@ func main() {
 		logger.WithError(err).Fatal("Error creating plank controller.")
 	}
 
+	// Push metrics to the configured prometheus pushgateway endpoint.
+	pushGateway := configAgent.Config().PushGateway
+	if pushGateway.Endpoint != "" {
+		go metrics.PushMetrics("plank", pushGateway.Endpoint, pushGateway.Interval)
+	}
+	// serve prometheus metrics.
+	go serve()
```
Comment (cjwagner) [inline prow/cmd/plank/main.go]: Why do we need to serve the metrics from the controller if we are already pushing them to the push gateway?

Comment (kargakis) [reply prow/cmd/plank/main.go]: The gateway is optional in case you run prow in a locked-down env. We don't use it downstream.

Comment (kargakis) [reply prow/cmd/plank/main.go]: In any case we should be able to access metrics from prow using both ways (pull, push).

**Pull request merged**